File: java/dnd-character/src/main/java/DiceResult.java
import java.util.List;
import java.util.ArrayList;
import java.util.stream.Collectors;
import java.util.Comparator;

class DiceResult {
    private List<Integer> values = new ArrayList();

    void add(int value) {
        values.add(value);
    }

    DiceResult top(int limit) {
        values = values.stream()
            .sorted(Comparator.reverseOrder())
            .limit(limit)
            .collect(Collectors.toList());

        return this;
    }

    int sum() {
        return values.stream().mapToInt(x -> x).sum();
    }
}

File: javascript/dnd-character/dice-result.js
export class DiceResult {
    constructor(values) {
        this.values = values;
    }

    top(n) {
        this.values = this.values.sort((a, b) => b - a).slice(0, n);
        return this;
    }

    sum() {
        return this.values.reduce((a, b) => a + b);
    }
}
